vello_shaders/shader/pathtag_scan_csdldf.wgsl

```diff
@@ -24,69 +24,49 @@ let LG_WG_SIZE = 8u;
 let WG_SIZE = 256u;
 
 //For the decoupled lookback
-let FLAG_NOT_READY: u32 = 0;
-let FLAG_REDUCTION: u32 = 1;
-let FLAG_INCLUSIVE: u32 = 2;
-let FLAG_MASK: u32 = 3;
+let FLAG_NOT_READY = 0u;
+let FLAG_REDUCTION = 1u;
+let FLAG_INCLUSIVE = 2u;
+let FLAG_MASK = 3u;
 
 //For the decoupled fallback
-let MAX_SPIN_COUNT: u32 = 4;
-let LOCKED: u32 = 1;
-let UNLOCKED: u32 = 0;
+let MAX_SPIN_COUNT = 4u;
+let LOCKED = 1u;
+let UNLOCKED = 0u;
 
 var<workgroup> sh_broadcast: u32;
 var<workgroup> sh_lock: u32;
-var<workgroup> sh_scratch: array<array<u32, 5>, WG_SIZE>;
-var<workgroup> sh_fallback: array<array<u32, 5>, WG_SIZE>;
-var<workgroup> sh_tag_broadcast: array<u32, 5>;
-var<workgroup> sh_fallback_state: array<bool, 5>;
+var<workgroup> sh_scratch: array<array<u32, PATH_MEMBERS>, WG_SIZE>;
+var<workgroup> sh_tag_broadcast: array<u32, PATH_MEMBERS>;
+var<workgroup> sh_fallback_state: array<bool, PATH_MEMBERS>;
 
-fn attempt_lookback(
-    part_ix: u32,
-    lookback_ix: u32,
-    member_ix: u32,
-    aggregate: u32,
-    spin_count: ptr<function, u32>,
-    prev: ptr<function, u32>,
-    reduction_complete: ptr<function, bool>,
-    inclusive_complete: ptr<function, bool>
-){
-    let payload = atomicLoad(&reduced[lookback_ix][member_ix]);
-    let flag_value = payload & FLAG_MASK;
-    if flag_value == FLAG_REDUCTION {
-        *spin_count = 0u;
-        *prev += payload >> 2u;
-        *reduction_complete = true;
-    } else if flag_value == FLAG_INCLUSIVE {
-        *spin_count = 0u;
-        *prev += payload >> 2u;
-        atomicStore(&reduced[part_ix][member_ix], ((aggregate + *prev)  << 2u) | FLAG_INCLUSIVE);
-        sh_tag_broadcast[member_ix] = *prev;
-        *inclusive_complete = true;
-    }
+struct pathtag_wrapper{
+    p: array<u32, PATH_MEMBERS>
 }
 
-fn fallback(
-    part_ix: u32,
-    fallback_ix: u32,
-    member_ix: u32,
-    aggregate: u32,
-    fallback_aggregate: u32,
-    prev: ptr<function, u32>,
-    inclusive_complete: ptr<function, bool>
-){
-    let fallback_payload = (fallback_aggregate << 2u) | select(FLAG_INCLUSIVE, FLAG_REDUCTION, fallback_ix != 0u);
-    let prev_payload = atomicMax(&reduced[fallback_ix][member_ix], fallback_payload);
-    if prev_payload == 0u {
-        *prev += fallback_aggregate;
-    } else {
-        *prev += prev_payload >> 2u;
-    }
-    if fallback_ix == 0u || (prev_payload & FLAG_MASK) == FLAG_INCLUSIVE {
-        atomicStore(&reduced[part_ix][member_ix], ((aggregate + *prev)  << 2u) | FLAG_INCLUSIVE);
-        sh_tag_broadcast[member_ix] = *prev;
-        *inclusive_complete = true;
-    }
+struct state_wrapper{
+    s: array<bool, PATH_MEMBERS>
+}
+
+//TODO: There has to be a better way to initialize an array?
+fn clear_pathtag()->array<u32, PATH_MEMBERS>{
+    var a: array<u32, PATH_MEMBERS>;
+    a[0] = 0u;
+    a[1] = 0u;
+    a[2] = 0u;
+    a[3] = 0u;
+    a[4] = 0u;
+    return a;
```

Comment:
```
return array(0u, 0u, 0u, 0u, 0u);
```
should also work. I don't think there's a nice syntax for a zero-initialised array.